lib/ts/recipe/emailpassword/recipe.ts

```diff
@@ -101,6 +105,183 @@ export default class Recipe extends RecipeModule {
                 Recipe.instance = new Recipe(Recipe.RECIPE_ID, appInfo, isInServerlessEnv, config, {
                     emailDelivery: undefined,
                 });
+
+                PostSuperTokensInitCallbacks.addPostInitCallback(() => {
+                    const mfaInstance = MultiFactorAuthRecipe.getInstance();
+                    if (mfaInstance !== undefined) {
+                        mfaInstance.addGetAllFactorsFromOtherRecipesFunc((tenantConfig) => {
+                            if (tenantConfig.emailPassword.enabled === false) {
+                                return [];
+                            }
+                            return ["emailpassword"];
+                        });
+                        mfaInstance.addGetFactorsSetupForUserFromOtherRecipes(async (tenantId: string, user: User) => {
```

Comment:
there is an issue with how we filter out fake email for this factor cause if there is no other email in the login methods of the user, or if there is, but all are fake, then we will end up asking users to enter an email for emailpassword factor, and the issue with that is that it won't be verified which will prevent it from being linked.

Therefore, we should not filter out fake emails here if they exist (in this function and the next one). If no email exists, then we should perhaps generate a fake email and mark it as verified on its own in sign up API.

This is not the case for passwordless recipe though, cause in that, it is verified by the login method itself..

Reply:
@porcellus